js/modules/k6/experimental/tracing/sampling.go

```diff
@@ -0,0 +1,59 @@
+package tracing
+
+import (
+	"math/rand"
+)
+
+// Sampler is an interface defining probabilistic sampling strategies.
+type Sampler interface {
+	// ShouldSample returns true if the trace should be sampled
+	// false otherwise.
+	ShouldSample() bool
+}
+
+// ProbabilisticSampler implements the ProbabilisticSampler interface and allows
+// to take probabilistic sampling decisions based on a sampling rate.
+type ProbabilisticSampler struct {
+	// random is a random number generator used by the sampler.
+	random *rand.Rand
+
+	// samplingRate is a chance value defined as a percentage
+	// value within 0.0 <= samplingRate <= 1.0 bounds.
+	samplingRate float64
+}
+
+// NewProbabilisticSampler returns a new ProbablisticSampler with the provided sampling rate.
+func NewProbabilisticSampler(samplingRate float64) *ProbabilisticSampler {
+	return &ProbabilisticSampler{samplingRate: samplingRate}
```

Comment:
Should we validate or discard negative numbers here?

Reply:
Good call, they keep the number withing bounds in the opentelemetry implementation too. I've added bounds check, and will set any number `< 0.0` to `0.0`, and any number bigger than `1.0` to `1.0`.